src/EditorFeatures/Core.Wpf/NavigateTo/NavigateToItemProvider.Searcher.cs

```diff
@@ -84,19 +98,80 @@ private class Searcher
                 }
             }
 
-            private async Task SearchAsync(Project project)
+            private async Task SearchProjectsInPriorityOrder(IDocumentTrackingService docTrackingService)
+            {
+                var processedProjects = new HashSet<Project>();
+
+                var activeDocOpt = docTrackingService.GetActiveDocument(_solution);
+                var visibleDocs = docTrackingService.GetVisibleDocuments(_solution)
+                                                    .Where(d => d != activeDocOpt)
+                                                    .ToImmutableArray();
+
+                // First, if there's an active document, search that project first, prioritizing
+                // that active document and all visible documents from it.
+                if (activeDocOpt != null)
+                {
+                    var activeProject = activeDocOpt.Project;
+                    processedProjects.Add(activeProject);
+
+                    var visibleDocsFromProject = visibleDocs.Where(d => d.Project == activeProject);
+                    var priorityDocs = ImmutableArray.Create(activeDocOpt).AddRange(visibleDocsFromProject);
+
+                    // Search the active project first.  That way we can deliver results that are
+                    // closer in scope to the user quicker without forcing them to do something like
+                    // NavToInCurrentDoc
+                    await Task.Run(() => SearchAsync(activeProject, priorityDocs), _cancellationToken).ConfigureAwait(false);
```

Comment:
so, I guess _cancellation is missing in SearchAsync since it is a field? since priorityDocs are not determined by input to Search all, how about making priorityDocs as field as well, and then don't need to pass down on every call?

Reply:
the priority docs are different *per* project you are searching.  so it can't be a field.

Reply:
you changed the list I saw, but looks like that doesn't affect any result. you can give full list everything, and it will still work same.

Reply:
Sorry, i'm confused by this:

> you changed the list I saw, but looks like that doesn't affect any result. you can give full list everything, and it will still work same.

THe list is used to determine the order that documents are processed in.  If we pass the full list, then we're effectively saying that no documents should be processed first.  Can you clarify why you think "it will work the same" and won't actually be impacted here?  Thanks!

Reply:
okay, let me try again.

if you change code to this.
var list = activeDocOpt != null ? ImmutableArray.Create(activeDocOpt).AddRange(visibleDocs) : visibleDocs;

and pass the list to every place you have passed in priority documents list. 

and then in ProjectSearch(project, priorityDocuments, ... ) filer out any documents that is not part of the project given.

that way, one doesn't need to create new immutable array per project or do GroupBy and etc. the list can be created at the constructor once so no need to pass it around methods.

like I said, this doesn't change functionality. so if you believe current way is better. sure.

Reply:
Ah.  Thank you for the explanation. I see now what you mean.  Yes, i think such an approach could work.

However, my primary issue is simply that i think such an API is fairly confusing.  I would not want to have an API that says: "give priority to these docids in this project".  It makes it really feel like it's a bug that those docids could refer to something *outside* of the project.  It's similar to if an API took a Solution and a Project and hte Project was not from that Solution.

Technically, you could completely document it and say it was "by design", but i still think it's an intuitive thing to do in an API, and i'd prefer the API was crisp and clear here.

Thanks!